security-jwt/src/main/java/io/micronaut/security/token/jwt/cookie/JwtCookieLoginHandler.java

```diff
@@ -60,10 +60,11 @@ public JwtCookieLoginHandler(JwtCookieConfiguration jwtCookieConfiguration,
     public HttpResponse loginSuccess(UserDetails userDetails, HttpRequest<?> request) {
         Optional<AccessRefreshToken> accessRefreshTokenOptional = accessRefreshTokenGenerator.generate(userDetails);
         if (accessRefreshTokenOptional.isPresent()) {
-            Cookie cookie = Cookie.of(jwtCookieConfiguration.getCookieName(), accessRefreshTokenOptional.get().getAccessToken());
-            cookie.path(jwtCookieConfiguration.getCookiePath());
-            cookie.maxAge(jwtGeneratorConfiguration.getAccessTokenExpiration());
-            cookie.httpOnly(true).secure(request.isSecure());
+
+            Cookie cookie = Cookie.of(request, jwtCookieConfiguration, accessRefreshTokenOptional.get().getAccessToken());
+            if (!jwtCookieConfiguration.getCookieMaxAge().isPresent()) {
```

Comment:
Would this be considered a breaking change? Previously the `maxAge` was always set and now it is set conditionally.

Reply:
I don't think it is a breaking change. 

> Previously the maxAge was always set and now it is set conditionally. 

If max age is not set in configuration and it is not

https://github.com/micronaut-projects/micronaut-core/blob/cookie-configuration/security-jwt/src/main/java/io/micronaut/security/token/jwt/cookie/JwtCookieConfigurationProperties.java#L98

then same as before is used

`cookie.maxAge(jwtGeneratorConfiguration.getAccessTokenExpiration());`

> Same thing with httpOnly. Previously it was always set to true and now it reads from configuration

The default value in configuration is Http Only: 

https://github.com/micronaut-projects/micronaut-core/blob/cookie-configuration/security-jwt/src/main/java/io/micronaut/security/token/jwt/cookie/JwtCookieConfigurationProperties.java#L53